File: ckgame-java/src/main/java/com/ckgame/events/Storyline.java
package com.ckgame.events;

import java.util.ArrayList;
import java.util.List;

/**
 * 剧情线。
 */
public final class Storyline {
    public final int id;
    public final String title;
    public final String description;
    public final int characterId;
    public StorylineStatus status = StorylineStatus.AVAILABLE;
    public int currentStage = 0;
    public final List<StorylineStage> stages = new ArrayList<>();
    public int startYear = 0;
    public int endYear = 0;
    public final List<String> tags = new ArrayList<>();

    public Storyline(int id, String title, String description, int characterId) {
        this.id = id;
        this.title = title;
        this.description = description;
        this.characterId = characterId;
    }

    public boolean isActive() {
        return status == StorylineStatus.ACTIVE;
    }

    public boolean isCompleted() {
        return status == StorylineStatus.COMPLETED;
    }

    public boolean isFailed() {
        return status == StorylineStatus.FAILED;
    }
}

File: ckgame-ts/src/events/Storyline.ts
import { StorylineStatus } from './StorylineStatus.js';
import { StorylineStage } from './StorylineStage.js';

/**
 * A storyline (long-term narrative arc).
 */
export class Storyline {
  public readonly id: number;
  public readonly title: string;
  public readonly description: string;
  public readonly characterId: number;
  public status: StorylineStatus = StorylineStatus.AVAILABLE;
  public currentStage: number = 0;
  public stages: StorylineStage[] = [];
  public startYear: number = 0;
  public endYear: number = 0;
  public tags: string[] = [];

  constructor(id: number, title: string, description: string, characterId: number) {
    this.id = id;
    this.title = title;
    this.description = description;
    this.characterId = characterId;
  }

  isActive(): boolean {
    return this.status === StorylineStatus.ACTIVE;
  }

  isCompleted(): boolean {
    return this.status === StorylineStatus.COMPLETED;
  }

  isFailed(): boolean {
    return this.status === StorylineStatus.FAILED;
  }
}
